devdox/tests/unit_test/app/services/test_api_keys_service.py

```
@@ -0,0 +1,122 @@
+        fake_store.set_existing_hash(fake_manager.hash_key("dvd_mock_key"))
+        fake_manager.set_fixed_key("dvd_mock_key")
+
+        service = PostApiKeyService(api_key_store=fake_store, api_key_manager=fake_manager)
+        user_claims = DummyUserClaims("user-123")
+
+        with pytest.raises(BadRequest) as exc:
+            await service.generate_api_key(user_claims)
+
+        assert exc.value.user_message == UNIQUE_API_KEY_GENERATION_FAILED
+        
+        pattern = FAILED_GENERATE_API_KEY_RETRIES_LOG_MESSAGE.format(attempts = "\d+")
```

Flake8 (7.2.0)

[error] unexpected spaces around keyword / parameter equals

(E251)